airbyte-integrations/connector-templates/source-python/unit_tests/unit_test.py.hbs

```diff
@@ -22,6 +22,8 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 """
 
+# generated
```

Comment:
why are you adding this comment to some of these templates and not others?

Reply:
in https://github.com/airbytehq/airbyte/blob/21c29933734c7f079c9827f0ea48a80b0ff47fa2/build.gradle#L81

Spotless detects in python files the first `import` line to add LICENSE headers before (with 1 blank line in between)

in cases where there are no `import`, it fallbacks on other keywords... such as `def` (first function definition)

However in PEP, writing 1 blank line between the LICENSE and the def line is wrong, it should be two blank lines!!
Then when black formatting is run, it automatically adds two lines whereas spotless is removing one (going for one blank only)

if you run:

`./gradlew :airbyte-integrations:connectors:source-file:build --dry-run`

You get:

```
:airbyte-commons:compileJava SKIPPED
:tools:code-generator:compileJava SKIPPED
:tools:code-generator:processResources SKIPPED
:tools:code-generator:classes SKIPPED
:tools:code-generator:jar SKIPPED
:tools:code-generator:assemble SKIPPED
:tools:code-generator:airbyteDocker SKIPPED
:airbyte-integrations:bases:airbyte-protocol:generateProtocolClassFilesWithoutLicense SKIPPED
:spotlessPython SKIPPED
:spotlessPythonApply SKIPPED
:airbyte-integrations:connectors:source-file:checkPython SKIPPED
:airbyte-integrations:connectors:source-file:pipInstall SKIPPED
:airbyte-integrations:connectors:source-file:blackFormat SKIPPED
:airbyte-integrations:connectors:source-file:flakeCheck SKIPPED
:airbyte-integrations:bases:airbyte-protocol:checkPython SKIPPED
:airbyte-integrations:bases:airbyte-protocol:pipInstall SKIPPED
:airbyte-integrations:bases:airbyte-protocol:blackFormat SKIPPED
:airbyte-integrations:bases:airbyte-protocol:flakeCheck SKIPPED
:airbyte-integrations:bases:airbyte-protocol:generateProtocolClassFiles SKIPPED
:airbyte-integrations:bases:airbyte-protocol:installLocalReqs SKIPPED
:airbyte-integrations:bases:airbyte-protocol:installReqs SKIPPED
:airbyte-integrations:bases:airbyte-protocol:isortFormat SKIPPED
:airbyte-integrations:bases:airbyte-protocol:airbytePythonApply SKIPPED
:airbyte-integrations:bases:base-normalization:checkPython SKIPPED
:airbyte-integrations:bases:base-normalization:pipInstall SKIPPED
:airbyte-integrations:bases:base-normalization:blackFormat SKIPPED
:airbyte-integrations:bases:base-normalization:flakeCheck SKIPPED
:airbyte-integrations:bases:base-normalization:installLocalReqs SKIPPED
:airbyte-integrations:bases:base-normalization:installReqs SKIPPED
:airbyte-integrations:bases:base-normalization:isortFormat SKIPPED
:airbyte-integrations:bases:base-normalization:airbytePythonApply SKIPPED
:airbyte-integrations:bases:base-python:checkPython SKIPPED
:airbyte-integrations:bases:base-python:pipInstall SKIPPED
:airbyte-integrations:bases:base-python:blackFormat SKIPPED
:airbyte-integrations:bases:base-python:flakeCheck SKIPPED
:airbyte-integrations:bases:base-python:installLocalReqs SKIPPED
:airbyte-integrations:bases:base-python:installReqs SKIPPED
:airbyte-integrations:bases:base-python:isortFormat SKIPPED
:airbyte-integrations:bases:base-python:airbytePythonApply SKIPPED
:airbyte-integrations:bases:base-python-test:checkPython SKIPPED
:airbyte-integrations:bases:base-python-test:pipInstall SKIPPED
:airbyte-integrations:bases:base-python-test:blackFormat SKIPPED
:airbyte-integrations:bases:base-python-test:flakeCheck SKIPPED
:airbyte-integrations:bases:base-python-test:installLocalReqs SKIPPED
:airbyte-integrations:bases:base-python-test:installReqs SKIPPED
:airbyte-integrations:bases:base-python-test:isortFormat SKIPPED
:airbyte-integrations:bases:base-python-test:airbytePythonApply SKIPPED
:airbyte-integrations:bases:base-singer:checkPython SKIPPED
:airbyte-integrations:bases:base-singer:pipInstall SKIPPED
:airbyte-integrations:bases:base-singer:blackFormat SKIPPED
:airbyte-integrations:bases:base-singer:flakeCheck SKIPPED
:airbyte-integrations:bases:base-singer:installLocalReqs SKIPPED
:airbyte-integrations:bases:base-singer:installReqs SKIPPED
:airbyte-integrations:bases:base-singer:isortFormat SKIPPED
:airbyte-integrations:bases:base-singer:airbytePythonApply SKIPPED
:airbyte-integrations:connectors:source-file:installLocalReqs SKIPPED
:airbyte-integrations:connectors:source-file:installReqs SKIPPED
:airbyte-integrations:connectors:source-file:isortFormat SKIPPED
:airbyte-integrations:connectors:source-file:airbytePythonApply SKIPPED
:airbyte-integrations:connectors:source-file:installTestReqs SKIPPED
:airbyte-integrations:connectors:source-file:unitTest SKIPPED
:airbyte-integrations:connectors:source-file:airbytePythonTest SKIPPED
:airbyte-integrations:bases:base-python:installTestReqs SKIPPED
:airbyte-integrations:bases:base-python:unitTest SKIPPED
:airbyte-integrations:bases:base-python:airbytePythonTest SKIPPED
:airbyte-integrations:bases:airbyte-protocol:installTestReqs SKIPPED
:airbyte-integrations:bases:airbyte-protocol:unitTest SKIPPED
:airbyte-integrations:bases:airbyte-protocol:airbytePythonTest SKIPPED
:airbyte-integrations:bases:airbyte-protocol:compileJava SKIPPED
:airbyte-integrations:bases:airbyte-protocol:processResources SKIPPED
:airbyte-integrations:bases:airbyte-protocol:classes SKIPPED
:airbyte-integrations:bases:airbyte-protocol:jar SKIPPED
:airbyte-integrations:bases:airbyte-protocol:assemble SKIPPED
:airbyte-integrations:bases:airbyte-protocol:airbyteDocker SKIPPED
:airbyte-integrations:bases:base:compileJava SKIPPED
:airbyte-integrations:bases:base:processResources SKIPPED
:airbyte-integrations:bases:base:classes SKIPPED
:airbyte-integrations:bases:base:jar SKIPPED
:airbyte-integrations:bases:base:assemble SKIPPED
:airbyte-integrations:bases:base:airbyteDocker SKIPPED
:airbyte-integrations:bases:base-python:compileJava SKIPPED
:airbyte-integrations:bases:base-python:processResources SKIPPED
:airbyte-integrations:bases:base-python:classes SKIPPED
:airbyte-integrations:bases:base-python:jar SKIPPED
:airbyte-integrations:bases:base-python:assemble SKIPPED
:airbyte-integrations:bases:base-python:airbyteDocker SKIPPED
:airbyte-integrations:connectors:source-file:compileJava SKIPPED
:airbyte-integrations:connectors:source-file:processResources SKIPPED
:airbyte-integrations:connectors:source-file:classes SKIPPED
:airbyte-integrations:connectors:source-file:jar SKIPPED
:airbyte-integrations:connectors:source-file:assemble SKIPPED
:airbyte-integrations:connectors:source-file:airbyteDocker SKIPPED
:airbyte-commons:processResources SKIPPED
:airbyte-commons:classes SKIPPED
:airbyte-commons:jar SKIPPED
:airbyte-integrations:connectors:source-file:compileTestJava SKIPPED
:airbyte-integrations:connectors:source-file:processTestResources SKIPPED
:airbyte-integrations:connectors:source-file:testClasses SKIPPED
:airbyte-integrations:connectors:source-file:test SKIPPED
:airbyte-integrations:connectors:source-file:jacocoTestReport SKIPPED
:airbyte-integrations:connectors:source-file:check SKIPPED
:airbyte-integrations:connectors:source-file:build SKIPPED
```

Where spotless is applied on all python modules but black is applied only one the source-flle !!

So every python files that don't have any imports  will go spotless way with only one blank and you get dirty git repository on files you  never intended to touch

To avoid all that headache, I prefer the `# generated`  tag instead of `def`